components/brave_wallet/browser/cardano/cardano_create_transaction_task.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_CREATE_TRANSACTION_TASK_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_CREATE_TRANSACTION_TASK_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/memory/weak_ptr.h"
+#include "base/types/expected.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_rpc_schema.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_transaction.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+
+namespace brave_wallet {
+class CardanoWalletService;
+class CardanoTransaction;
+
+class CreateCardanoTransactionTask {
```

Comment:
Please add documentation to these classes/functions as instructed in the style guide https://google.github.io/styleguide/cppguide.html#Comments

Reply:
also are there some external Cardano docs you can link to for these classes? I think it would be really helpful to understand this stuff.

Reply:
Added a bit more comments.
With regards to comments: everything from `base` folder is designed reusable and to be seen by thousands of devs so these classes should be well documented. 
Internals of wallet are to be seen by far less amount of devs and it just not worth of time spend for it to be documented with such details.

Reply:
It is better to ask question, rather than assuming wrong how things are done. I have already mentioned this in another thread. Me and @bridiver have also had a different thread in another PR where we have explained that that the requirements for wallet are no different than for the rest of the codebase.

Reply:
Comments are better now